data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/GeoIPProcessorConfig.java

```diff
@@ -44,4 +47,11 @@ public List<String> getTagsOnFailure() {
         return tagsOnFailure;
     }
 
+    /**
+     * Get when condition
+     * @return String When condition
+     */
+    public String getWhenCondition() {
```

Comment:
Does it make sense to add this when condition to the entry level? Just checking. I'm asking because processors like `add_entries` has it on the entry level: https://github.com/opensearch-project/data-prepper/blob/2be8166ac8ac1f80784201f2a6e33302145cb5a8/data-prepper-plugins/mutate-event-processors/src/main/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorConfig.java#L31

Reply:
That's a good suggestion, I actually looked into it if we need it but I think it's worth making that change. I will move it next PR so it can be configured for each entry.